js/pages/characterizations/services/CharacterizationService.js

```diff
@@ -44,7 +44,16 @@ define([
     function loadCharacterizationExecutionList(id) {
         return httpService
             .doGet(config.webAPIRoot + 'cohort-characterization/' + id + '/generation')
-            .then(res => res.data);
+            .then(res => generateVersionTags(res.data));
+    }
+
+
+    function generateVersionTags(sourceInfoList) {
+        const sortedIds = sourceInfoList.map(info => info.id).sort();
```

Comment:
Probably sort descending by date is more suitable, since look what I got below:
![88bfd983-86f8-4061-9f39-b41991535579](https://user-images.githubusercontent.com/1109753/58547381-ecc58a00-820f-11e9-9534-b736e64a6211.png)

Reply:
Fixed - version generated according to distinct hashCode index